addons/pos_mercado_pago/static/src/app/payment_mercado_pago.js

```diff
@@ -0,0 +1,188 @@
+                this.env.services.orm.silent
+                    .call("pos.payment.method", "mp_payment_intent_cancel", [
+                        [line.payment_method.id],
+                        this.payment_intent.id,
+                    ])
+                    .then((result) => {
+                        resolve(result);
+                    });
+            });
+    }
+
+    async send_payment_request(cid) {
+        await super.send_payment_request(...arguments);
+        const line = this.pos.get_order().selected_paymentline;
+        try {
+            // During payment creation, user can't cancel the payment intent
+            line.set_payment_status("waitingCapture");
+            // Call Mercado Pago to create a payment intent
+            const payment_intent = await this.create_payment_intent();
+            if (!("id" in payment_intent)) {
+                this._showMsg(payment_intent.message, "error");
+                return false;
+            }
+            // Payment intent creation successfull, save it
+            this.payment_intent = payment_intent;
+            // After payment creation, make the payment intent canceling possible
+            line.set_payment_status("waitingCard");
+            // Wait for payment intent status change and return status result
+            return await Promise.any([
+                new Promise((resolve) => {
+                    this.webhook_resolver = resolve;
+                }),
+                new Promise((resolve) => {
+                    this.payment_cancel_resolver = resolve;
+                }),
+            ]);
+        } catch (error) {
+            this._showMsg(error, "System error");
+            return false;
+        }
+    }
+
+    async send_payment_cancel(order, cid) {
+        await super.send_payment_cancel(order, cid);
+        if (!("id" in this.payment_intent)) {
+            return true;
+        }
+        const canceling_status = await this.cancel_payment_intent();
+        if ("error" in canceling_status) {
+            const message =
+                canceling_status.status === 409
+                    ? _t("Payment has to be canceled on terminal")
+                    : _t("Payment not found (canceled/finished on terminal)");
+            this._showMsg(message, "info");
+            return canceling_status.status !== 409;
+        }
+        return true;
+    }
+
+    async handleMercadoPagoWebhook() {
+        const line = this.pos.get_order().selected_paymentline;
+        const MAX_RETRY = 5; // Maximum number of retries for the "ON_TERMINAL" BUG
+        const RETRY_DELAY = 1000; // Delay between retries in milliseconds for the "ON_TERMINAL" BUG
+        const showMessageAndResolve = (messageKey, status, resolverValue) => {
+            if (!resolverValue) {
+                this._showMsg(_t(messageKey), status);
+            }
+            line.set_payment_status("done");
+            this.webhook_resolver(resolverValue);
```

Comment:
`webhook_resolver` can be `null`.
Steps:
1. send payment request.
2. refresh the page.
3. when receiving the notification `handleMercadoPagoWebhook` will be called with `webhook_resolver = null`

How about only calling it when it's present?
```suggestion
            this.webhook_resolver?.(resolverValue);
```

Besides, when refreshing the page, there is no function call that's waiting for the `send_payment_request` to resolve. Setting the payment status is enough for the UI to adapt.